In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from plotly.offline import download_plotlyjs, init_notebook_mode, plot, iplot
import plotly.express as px
import plotly.graph_objects as go
import plotly.figure_factory as ff
from plotly.colors import n_colors
from plotly.subplots import make_subplots
init_notebook_mode(connected = True)
import cufflinks as cf
cf.go_offline()

In [6]:
victims = pd.read_csv('C:/workspace/Data/crime-in-india/20_Victims_of_rape.csv')
police_hr = pd.read_csv('C:/workspace/Data/crime-in-india/35_Human_rights_violation_by_police.csv')
auto_theft = pd.read_csv('C:/workspace/Data/crime-in-india/30_Auto_theft.csv')
prop_theft = pd.read_csv('C:/workspace/Data/crime-in-india/10_Property_stolen_and_recovered.csv')

## 강간 사건
- 강간은 인도에서 네번째로 많은 범죄율을 기록한다. 강간금지법은 인도 형법 376에 따른다. 근친상간사건은 가해자가 피해자를 인지하고 있다는 조건하에 등록이된다 (??)
    * 원문 : Incest rape cases are registered under the condition where the offender is known to the victim.
- 다음은 2001 ~ 2010 년도 인도에서 일어난 근친상간사건의 통계치이다

In [22]:
incest = victims.loc[victims['Subgroup'] == 'Victims of Incest Rape']
incest = incest.groupby('Year').sum()
incest

Rape_Cases_Reported  Victims_Above_50_Yrs  Victims_Between_10-14_Yrs  \
Year                                                                         
2001                  439                     1                         63   
2002                  369                     5                         58   
2003                  399                     0                         41   
2004                  505                     1                         43   
2005                  750                     3                         64   
2006                  431                     3                         55   
2007                  405                     2                         45   
2008                  309                     8                         54   
2009                  321                     2                         31   
2010                  288                     1                         59   

      Victims_Between_14-18_Yrs  Victims_Between_18-30_Yrs  \
Year                                                         
2001                        112                        170   
2002                         77                        176   
2003                         99                        206   
2004                        134                        250   
2005                        164                        357   
2006                        112                        182   
2007                        114                        167   
2008                         75                        117   
2009                         76                        158   
2010                         79                         99   

      Victims_Between_30-50_Yrs  Victims_of_Rape_Total  Victims_Upto_10_Yrs  
Year                                                                         
2001                         69                    439                   24  
2002                         38                    369                   15  
2003                         31                    400                   23  
2004                         66                    506                   12  
2005                        126                    750                   36  
2006                         55                    431                   24  
2007                         49                    405                   28  
2008                         32                    309                   23  
2009                         31                    322                   24  
2010                         20                    288                   30

In [26]:
fig = px.bar(incest, x = incest.index, y = 'Rape_Cases_Reported', title = 'Total Number of Cases of Incest Rape')
fig.show()

- 2005년 750회로 정점을 찍고 점점 감소하는 추세를 보인다
- 최저점은 2010년으로 총 288회